test/test_eager_transforms.py

```diff
@@ -279,6 +279,14 @@ def test_argnums(self, device):
         self.assertEqual(gx, y)
         self.assertEqual(gy, x)
 
+    def test_empty_grad(self, device):
+        def f(x):
+            return (x['a']**2.0).sum()
+        inps = ({'a':torch.randn(10) + 3, 'b':torch.randn(10)})
+        grads = grad(f)(inps)
+        self.assertNotEqual(grads['a'].sum(), 0.0)
+        self.assertEqual(grads['b'].sum(), 0.0)
+
```

Comment:
Could we add a test case for https://github.com/zou3519/functorch/issues/1 as well? I think it should pass with this change (but if there is something else wrong it does not need to be handled in this PR)

Reply:
Unfortunately, vjp uses its own call and doesn't use grad.